static/app/views/detectors/components/detectorLink.tsx

```diff
@@ -1,43 +1,183 @@
 
 import ProjectBadge from 'sentry/components/idBadge/projectBadge';
 import Link from 'sentry/components/links/link';
 import {IconSentry} from 'sentry/icons';
+import {t} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
+import type {DataCondition} from 'sentry/types/workflowEngine/dataConditions';
+import {
+  DataConditionType,
+  DetectorPriorityLevel,
+} from 'sentry/types/workflowEngine/dataConditions';
+import type {DataSource, Detector} from 'sentry/types/workflowEngine/detectors';
+import {defined} from 'sentry/utils';
+import getDuration from 'sentry/utils/duration/getDuration';
+import {middleEllipsis} from 'sentry/utils/string/middleEllipsis';
+import {unreachable} from 'sentry/utils/unreachable';
 import useOrganization from 'sentry/utils/useOrganization';
 import useProjectFromId from 'sentry/utils/useProjectFromId';
 import {makeMonitorDetailsPathname} from 'sentry/views/detectors/pathnames';
 
 type DetectorLinkProps = {
-  createdBy: string | null;
-  detectorId: string;
-  disabled: boolean;
-  name: string;
-  projectId: string;
+  detector: Detector;
   className?: string;
 };
 
-export function DetectorLink({
-  detectorId,
-  name,
-  createdBy,
-  projectId,
-  className,
-  disabled,
-}: DetectorLinkProps) {
+function formatConditionPriority(condition: DataCondition) {
+  switch (condition.conditionResult) {
+    case DetectorPriorityLevel.HIGH:
+      return 'high';
+    case DetectorPriorityLevel.MEDIUM:
+      return 'medium';
+    default:
+      return 'low';
+  }
+}
+
+function formatConditionType(condition: DataCondition) {
+  switch (condition.type) {
+    case DataConditionType.GREATER:
+      return '>';
+    case DataConditionType.LESS:
+      return '<';
+    case DataConditionType.EQUAL:
+      return '=';
+    case DataConditionType.NOT_EQUAL:
+      return '!=';
+    case DataConditionType.GREATER_OR_EQUAL:
+      return '>=';
+    case DataConditionType.LESS_OR_EQUAL:
+      return '<=';
+    default:
+      return condition.type;
+  }
+}
+
+function formatCondition({condition, unit}: {condition: DataCondition; unit: string}) {
+  if (
+    !condition.conditionResult ||
+    condition.conditionResult === DetectorPriorityLevel.OK
+  ) {
+    return null;
+  }
+
+  const comparison = formatConditionType(condition);
+  const threshold = `${condition.comparison}${unit}`;
+  const priority = formatConditionPriority(condition);
+
+  return `${comparison}${threshold} ${priority}`;
+}
+
+function DetailItem({children}: {children: React.ReactNode}) {
+  if (!children) {
+    return null;
+  }
+
+  return (
+    <Fragment>
+      <Separator />
+      <DetailItemContent>{children}</DetailItemContent>
+    </Fragment>
+  );
+}
+
+function ConfigDetails({detector}: {detector: Detector}) {
+  const type = detector.config.detection_type;
+  const conditions = detector.conditionGroup?.conditions;
+  if (!conditions?.length) {
+    return null;
+  }
+
+  switch (type) {
```

Comment:
rather than building these with switch statements, could we make a constants file that exports an object that we can use to lookup these values?

Reply:
Yeah we certainly can and probably will do something like that as we build out the UI. I'm waiting until we get some of the detector details page built out so we can see if there are any abstractions we can take advantage of. Definitely want to make it simple for future developers to make new detector types